connector/exceptionsconnector/connector_logs.go

```diff
@@ -0,0 +1,132 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exceptionsconnector
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+)
+
+type logsConnector struct {
+	logsConsumer consumer.Logs
+	component.StartFunc
+	component.ShutdownFunc
+
+	logger *zap.Logger
+	ld     plog.Logs
+}
+
+func newLogsConnector(logger *zap.Logger) (*logsConnector, error) {
+	logger.Info("Building logs exceptionsconnector")
+
+	return &logsConnector{
+		logger: logger,
+		ld:     plog.NewLogs(),
+	}, nil
+}
+
+// Capabilities implements the consumer interface.
+func (c *logsConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements the consumer.Traces interface.
+// It aggregates the trace data to generate logs.
+func (c *logsConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	c.logger.Debug("Consume traces")
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		rspans := traces.ResourceSpans().At(i)
+		resourceAttr := rspans.Resource().Attributes()
+		serviceAttr, ok := resourceAttr.Get(conventions.AttributeServiceName)
+		if !ok {
+			continue
+		}
+		serviceName := serviceAttr.Str()
+		ilsSlice := rspans.ScopeSpans()
+		for j := 0; j < ilsSlice.Len(); j++ {
+			ils := ilsSlice.At(j)
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				for l := 0; l < span.Events().Len(); l++ {
+					event := span.Events().At(l)
+					if event.Name() == "exception" {
+						c.attrToLogRecord(serviceName, span.Attributes(), event.Attributes())
+					}
+				}
+			}
+		}
+	}
+	c.exportLogs(ctx)
+	return nil
+}
+
+func (c *logsConnector) exportLogs(ctx context.Context) error {
+	c.logger.Debug("Exporting logs")
+	if err := c.logsConsumer.ConsumeLogs(ctx, c.ld); err != nil {
+		c.logger.Error("Failed ConsumeLogs", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+func (c *logsConnector) attrToLogRecord(serviceName string, spanAttr, eventAttr pcommon.Map) plog.LogRecord {
+	rl := c.ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	logRecord := sl.LogRecords().AppendEmpty()
+
+	logRecord.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
```

Comment:
I think this is technically incorrect. The `Timestamp` field should reflect the time at which the event occurred. In the case of translating from spans, I think that means we need to show the start or end time (perhaps configurable in the future?). In any case, the `ObservedTimestamp` will be automatically set to `time.Now`, and this can be used to understand when the log was created.